pkg/apiserver/registry/clusterpedia/resources/rest_test.go

```diff
@@ -0,0 +1,149 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	genericrequest "k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/clusterpedia-io/clusterpedia/pkg/utils/request"
+)
+
+const testResponseTmpl = "Cluster Name: %s, Proxy Path: %s"
+
+func newResponseBody(cluster string, proxyPath string) string {
+	return fmt.Sprintf(testResponseTmpl, cluster, proxyPath)
+}
+
+type testHandler struct{}
+
+func (*testHandler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+	cluster := request.ClusterNameValue(req.Context())
+	_, _ = writer.Write([]byte(newResponseBody(cluster, req.URL.Path)))
+}
+
+// responder implements rest.Responder for assisting a connector in writing objects or errors.
+type responder struct {
+	w http.ResponseWriter
+}
+
+func (r *responder) Object(statusCode int, obj runtime.Object) {}
+
+func (r *responder) Error(err error) {
+	if status, ok := err.(apierrors.APIStatus); ok {
+		r.w.WriteHeader(int(status.Status().Code))
+	}
+	_, _ = r.w.Write([]byte(err.Error()))
+}
+
+func newTestRequestInfoResolver() *genericrequest.RequestInfoFactory {
+	return &genericrequest.RequestInfoFactory{
+		APIPrefixes:          sets.NewString("api", "apis"),
+		GrouplessAPIPrefixes: sets.NewString("api"),
+	}
+}
+
+func TestConnect(t *testing.T) {
+	tests := []struct {
+		url string
+
+		expectedCode int
+		expectedBody string
+	}{
+		{
+			"/api/v1/resources",
+
+			200,
+			newResponseBody("", ""),
```

Comment:
This doesn't look very intuitive, so we can use `expectedCluster` and `expectedURL` and generate the response body in the test